lib/actions/apiV2.js

```diff
@@ -1105,17 +1105,16 @@ export function routingQuery(searchId = null, updateSearchInReducer) {
                 returned even if trips were returned, since it is on a mode-by-mode basis.
                 there is a chance for user confusion! 
                 
-                By checking if itineraries exist, we can hide this error when it is
-                not applicable. */
+               We'll reintroduce this error later once all the results are compiled */
+
+              response.data.plan.routingErrors =
+                response.data?.plan?.routingErrors.filter(
+                  (re) => re?.code !== 'NO_TRANSIT_CONNECTION'
+                )
               if (
                 getActiveItineraries(getState())?.length > 0 &&
                 response?.data?.plan
               ) {
-                response.data.plan.routingErrors =
-                  response.data?.plan?.routingErrors.filter(
-                    (re) => re?.code !== 'NO_TRANSIT_CONNECTION'
-                  )
-
```

Comment:
I'm a bit confused as to why this was moved. I believe this was inside this block to prevent a crash when there was an empty result from the server?

Reply:
This is causing the `NO_TRANSIT_CONNECTION` error in cases where there aren't walk/transit combinations! I think it's just cleaner to fully avoid having the error unless we know we want to. I did some testing for this PR with trip searches that return 0 itineraries and it seemed to be working okay, but let me know if there's something specific I can test!

Reply:
Great! I can't remember anymore. I played with it a bunch and nothing crashed but maybe we should add a comment saying that this might cause weird behavior?